csrc/scheduler/normalization.cpp

```diff
@@ -389,9 +388,9 @@ std::shared_ptr<ReductionParams> innerPersistentHeuristic(
   // block by a factor of 2
   if (batches_per_block_outer_reduction * batches_per_block_inner_reduction *
               inner_reduction_unroll_factor * outer_reduction_unroll_factor *
-              4 >
-          255 * 3 &&
-      bdimx * bdimy * bdimz * 2 <= max_threads_in_block &&
+              4ll >
+          255ll * 3ll &&
+      bdimx * bdimy * bdimz * 2ll <= max_threads_in_block &&
```

Comment:
When I removed `255ll *3ll`, I see these errors:
> Lint for csrc/scheduler/normalization.cpp:

 Error (CLANGTIDY) [bugprone-implicit-widening-of-multiplication-result,-warnings-as-errors] performing an implicit widening conversion to type 'long long' of a multiplication performed in type 'int'

         389  |  if (batches_per_block_outer_reduction * batches_per_block_inner_reduction *
         390  |              inner_reduction_unroll_factor * outer_reduction_unroll_factor *
         391  |              4ll >
    >>>  392  |          255 * 3 &&
         393  |      bdimx * bdimy * bdimz * 2ll <= max_threads_in_block &&
         394  |      batches_per_block_inner_reduction >
         395  |          batches_per_block_inner_reduction_max) {

Error (CLANGTIDY) [bugprone-narrowing-conversions,cppcoreguidelines-narrowing-conversions,-warnings-as-errors] narrowing conversion from 'int64_t' (aka 'long') to signed type 'int' is implementation-defined

         490  |
         491  |  auto rparams = std::make_shared<ReductionParams>();
         492  |
    >>>  493  |  rparams->cparams.maxrregcount = nvrtc_register_per_thread;
         494  |  rparams->persistent_kernel = true;
         495  |  rparams->fastest_dim = true;
         496  |

Reply:
The double `l` isn't needed here.

Reply:
Yes, I understand why clang-tidy complains and why `255ll * 3ll` is a workaround. It just feels bizarre that we can't just write `255 * 3` since it's just a simple integer math expression. Maybe this is a price to pay to use clang-tidy as it certainly has real benefits.

Reply:
It seems annoying that the default integer types in the cpp standard are `size_t` and `int64_t`, but default type for constants is `int`.